flashinfer/comm/trtllm_moe_alltoall.py

```
@@ -0,0 +1,743 @@
+    This class provides the throughput-optimized backend that supports multiple payloads
+    per collective operation, explicit dispatch/combine phases, and workspace-backed tensors.
+
+    Example:
+        >>> moe_a2a = MoeAlltoAll(mapping, max_num_tokens=2048, top_k=2, num_experts=8)
+        >>> recv = moe_a2a.dispatch(experts, [hidden, ids, scales], batch_size)
+        >>> output = moe_a2a.combine(processed, batch_size)
+    """
+
+    # Single shared workspace across the process
+    # _WORKSPACE: Optional[dict] = None
+    _WORKSPACE_CACHE: dict[tuple[int, int, int, int], dict] = {}
```

Ruff (0.14.8)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)